src/components/map/Map.tsx

```diff
@@ -73,6 +76,24 @@ export const Map = (props: MapProps) => {
     });
   };
 
+  useEffect(() => {
+    if (debouncedState) {
+      loadMobility({
+        visibleBounds: [
+          debouncedState.properties.bounds.ne,
+          debouncedState.properties.bounds.sw,
+        ],
+        zoomLevel: debouncedState.properties.zoom,
+        center: debouncedState.properties.center,
+      });
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
With the suggestion above, maybe it will work when wrapping `onMapIdle` and `loadMobility` with `useCallback`?

Reply:
Unfortunately, no, it still triggers multiple debounce due to location change if we put the dependency to the useEffect.